go/private/rules/cgo.bzl

```diff
@@ -345,13 +346,20 @@ def _cgo_codegen_impl(ctx):
                 # us the static variant. We'll get one file for each transitive dependency,
                 # so the same file may appear more than once.
                 if (lib.basename.startswith("lib") and
-                    any([lib.basename.endswith(ext) for ext in SHARED_LIB_EXTENSIONS])):
+                    has_shared_lib_extension(lib.basename)):
```

Comment:
Please also update the corresponding logic in `cgo_configure`. That's the new path. I hope to delete `_cgo_codegen_impl` and other rules soon, but both paths need to be supported.